packages/create/src/cli/change-data-source/change-data-source.ts

```diff
@@ -0,0 +1,106 @@
+import type { Command } from 'commander';
+import type { AdpProjectData, DataSourceData, PromptDefaults } from '@sap-ux/adp-tooling';
+import {
+    generateChange,
+    ChangeType,
+    ChangeDataSourcePrompts,
+    getTargetDataSources,
+    getTargetODataAnnotations,
+    getDataSourcesDictionary,
+    getDataServicesWithURI
+} from '@sap-ux/adp-tooling';
+import { prompt } from 'inquirer';
+import { createForAbap } from '../../../../axios-extension';
+import { readFileSync } from 'fs';
+import { join } from 'path';
+import { UI5Config } from '@sap-ux/ui5-config';
+
+/**
+ * Add a new sub-command to change the data source of an adaptation project to the given command.
+ *
+ * @param {Command} cmd - The command to add the change-data-source sub-command to.
+ */
+export function addChangeDataSourceCommand(cmd: Command): void {
+    cmd.command('change-data-source')
+        .argument('<path>', 'path to the Adaptation Project folder')
+        .requiredOption('--url [url]', 'url pointing to the target system containing the original app')
+        .requiredOption('--username [username]', 'username for the target system containing the original app')
```

Comment:
Please don't ask allow providing the password as CLI parameter. Most proper shells store these inputs in a plain history file.